Problem4/src/problem4/Problem4.java

```diff
@@ -10,12 +10,35 @@
  * @author cgallinaro
  */
 public class Problem4 {
-
+    
+    
     /**
-     * @param args the command line arguments
+     * Runtime Complexity : O(n^2)
+     * @param data 
      */
-    public static void main(String[] args) {
-        // TODO code application logic here
+    public static void sort(int[] data) {
+        
+        //loop over all entries backwards
+        for (int i = data.length - 1; i >= 1; i--) {
+            
+            //store the best value and best index
+            int best_index = -1;
+            int best_value = -1;
```

Comment:
If your array contains integers < -1, the initial value you are using will not work.  When testing your algorithm, I used the array [-2, -5, -4].  Your algorithm threw an exception because it tried to access index -1 in your array.